packages/dds/map/src/directory.ts

```diff
@@ -661,16 +662,13 @@ export class SharedDirectory extends SharedObject<ISharedDirectoryEvents> implem
      * {@inheritDoc @fluidframework/shared-object-base#SharedObject.loadCore}
      */
     protected async loadCore(storage: IChannelStorageService) {
-        const blob = await storage.readBlob(snapshotFileName);
-        const header = bufferToString(blob, "utf8");
-        const data = JSON.parse(header);
+        const data = await readAndParse(storage, snapshotFileName);
```

Comment:
I'd add assert here that data !== undefined, as semantics changed. readBlob() would throw, but readAndParse returns undefined.

Reply:
See comment below, I think nothing needs to be done here.